### PR TITLE
Update menu items to flat structure

### LINKED ISSUE #34: Menu items should be stored in a flat structure
Currently the menu items are stored in a tree structure, but as discussed in the main p2 thread, a flat structure is probably more advantageous in the front-end for finding and parsing information.

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #34 

* Changes data to flat from tree structure.
* Removes the `icon` property.
* Adds some initial tests around the menu.

### Testing

1. Make sure menu items still show as expected on the front-end.
1. Make sure tests pass.

### CODE REVIEW

File: tests/menu.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class WC_Tests_Navigation_Menu extends WC_REST_Unit_Test_Case {
+	/**
+	 * Test that menu items can be added.
+	 */
+	public function test_add_menu_items() {
+		Menu::add_category(
+			'Test Category',
+			'manage_woocommerce',
+			'test-category',
+			'',
```
Comment (mattsherman) [inline tests/menu.php:22]: Trailing commas are only supported in PHP 7.3 or greater.

File: tests/menu.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		);
+
+		Menu::add_item(
+			'test-category',
+			'Test Item',
+			'manage_woocommerce',
+			'test-item',
+			'',
```
Comment (mattsherman) [inline tests/menu.php:30]: Trailing commas are only supported in PHP 7.3 or greater.

File: tests/menu.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	/**
+	 * Test adding a menu item that has already been added.
+	 */
+	public function test_duplicate_menu_slug() {
+		Menu::add_item(
+			'test-category',
+			'Test Item',
+			'manage_woocommerce',
+			'test-item',
+			'',
```
Comment (mattsherman) [inline tests/menu.php:55]: Trailing commas are only supported in PHP 7.3 or greater.

File: tests/menu.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+	/**
+	 * Test that the callback is converted to a URL.
+	 */
+	public function test_callback_url() {
+		Menu::add_category(
+			__( 'Test Page', 'woocommerce-navigation' ),
+			'manage_woocommerce',
+			'test-plugins',
+			'wc-test',
```
Comment (mattsherman) [inline tests/menu.php:69]: Trailing commas are only supported in PHP 7.3 or greater.

**Comment by joshuatf** [comment]
>Unit tests fail due to trailing commas in function parameters. This is only supported in PHP 7.3 or greater.

Thanks for catching that, @mattsherman!  Updated in https://github.com/woocommerce/navigation/pull/35/commits/cbbd807fadb12a999d45efe13d082825f5a01920

I'm a little surprised our code sniffers don't throw any warnings about this.

**Comment by mattsherman** [comment]
> I'm a little surprised our code sniffers don't throw any warnings about this.

It looks like we are excluding `tests/` from `PHPCompatibility` checking -- I don't think that is probably best. Our tests should run on any version of PHP that we support, and we should check for that. I've created an issue and assigned it to myself to correct this: https://github.com/woocommerce/navigation/issues/41

**Pull request merged**